yo1k/tic_tac_toe/core/game.py

```diff
@@ -198,48 +187,39 @@ def advance(self, state: State):
         else:
             assert False
 
-    @staticmethod
-    def __turn(state: State) -> int:
-        """Returns `turn` - the index of the current `Player` in `State.players`.
-
-        `turn` is calculated in such a way as to alternate players order in different
-        `round` while keeping the same `Player` index in the `State.players`.
-        """
-        return (state.step + state.round) % 2
-
     @staticmethod
     def __occupy(state: State, cell: Cell):
         assert state.phase is Phase.INROUND
-        turn = Logic.__turn(state)
-        mark = state.players[turn].mark()
-        if state.board._get(cell) is None:
-            state.board._set(cell, mark)
+        turn = state.turn()
+        mark = state.players[turn].mark
+        if state.board.get(cell) is None:
+            state.board.set(cell, mark)
         else:
             assert False
-        if Logic._win_condition(state.board, cell):
+        if Logic.win_condition(state.board, cell):
             Logic.__win(state)
         elif Logic.__last_step(state.step):
             Logic.__draw(state)
         else:
             state.step += 1
 
     @staticmethod
-    def _win_condition(board: Board, last_occupied: Cell) -> bool:
+    def win_condition(board: Board, last_occupied: Cell) -> bool:
```

Comment:
So tests also can't access protected program elements? This is the only possible reason I can think of behind making this method public.

Reply:
Done.